### ISSUE TITLE
createToken using only sandbox API URL?

### ISSUE THREAD

**Issue by volomike** [payeezy/payeezy_js#24]
Question. Although I was doing okay (after a lot of trial/error and reading anything I could find on the web) and managed to get sandbox token-based transactions running for purchases and refunds, I was having a tough time getting transactions to work in production. The problems would stem from before even doing the full form post -- it was being returned by the auth step in Javascript. So, I took a look at this file:

https://github.com/payeezy/payeezy_js/blob/master/javascript/payeezy_us_v5.1.js

Inside, to my surprise, it looks like the URL where it submits data looks like a sandbox URL because it said **api-cert.payeezy.com** instead of **api.payeezy.com**. On a wild hunch, I switched it and ran more tests. To my wild surprise, this seemed to work finally and now I was getting past the Javascript check and the form submitted to the processor. I see the transactions in the Payeezy Gateway as auth 0.00 amount, but no charges yet.

So, my question to you is -- when we go into production, are we supposed to switch this line in the Javascript:

`var n = "api-cert.payeezy.com";`

...to this line?

`var n = 'api.payeezy.com';`

Now, second, I think the reason it is failing to charge on a live merchant account were two reasons:

1- On an AMEX card, it seemed to only like a card type spelled out as "American Express" (note the Proper Case) instead of "amex" or "american express". Odd! (I've already mentioned this in another issue.) Then, it finally returned a response:

`...{"code":"37","description":"GATEWAY:Payment Type Not Supported by Merchant"}...`

So, I think my biz partner may not have turned on AMEX processing, I'm guessing. I'll speak to him and then PayEezy later to verify.

2- On a MasterCard, it was a valid card, but said Invalid Credit Card Number with an error code of 22. It showed up properly as an auth 0.00 (no charge) but then failed to charge. I strongly suspect that the problem there is that I didn't use the proper Transarmor Token. (In this doc that I found (https://github.com/payeezy/get_started_with_payeezy), it says I need to get a proper ta_token per merchant account. Yet another key/token of sorts that I have to add -- we're now running 5 key/tokens just to do an auth and a charge, unlike Stripe which only needs 2.) I had my ta_token set to NOIW. I'm trying to contact my business partner to ask him to login to the portal as admin (I can't do this -- I'm just a power user, not admin) and then go to Terminals and retrieve the ta_token. He's not around at the moment.

**Comment by volomike** [comment]
Yep, I finally got a copy of the PayEezy.js Integration Guide PDF. When I read it, although it's not clearly stated, it says that we need to switch that URL just as I stated above.

Probably a better way to handle this would have been if the coder at PE had written this such that if I do Payeezy.setSandbox(true), it used the api-cert.payeezy.com, and Payeezy.setSandbox(false), it used the api.payeezy.com.

**Comment by rohitrajagopal** [comment]
@volomike - Thanks for the feedback. We are working to add a setEnvironment() function.

@renjy01

**Comment by CurvesTech** [comment]
@rohitrajagopal it's june 2017 and it's still not added.
